src/service.rs

```diff
@@ -54,7 +54,8 @@ impl<UID: Uid> Service<UID> {
         };
         MappingContext::new(options)
         .map_err(|e| CrustError::NatError(e))
-        .map(move |mc| {
+        .map(move |mut mc| {
+            mc.add_peer_stuns(config.read().hard_coded_contacts.iter().cloned());
```

Comment:
This commit seems like an important one and I guess it's related to warnings.
It would have been nice, if it was comited separately.
Just a note for the future :)

Reply:
You're right, that was just laziness.